apps/explorer/src/pages/id-page/index.tsx

```diff
@@ -11,39 +11,56 @@ import {
 import { PageLayout } from '~/components/Layout/PageLayout';
 import { PageHeader } from '~/ui/PageHeader';
 import { ObjectDetailsHeader } from '@mysten/icons';
-import { TotalStaked } from '~/pages/address-result/TotalStaked';
+import { TotalStaked } from './TotalStaked';
 import { ErrorBoundary } from '~/components/error-boundary/ErrorBoundary';
 import { ObjectView } from '~/pages/object-result/views/ObjectView';
-import { PageContent } from '~/pages/id-page/PageContent';
+import { PageContent } from './PageContent';
+import { type DataType, translate } from '~/pages/object-result/ObjectResultType';
+import { PackageDetails } from '~/pages/id-page/PackageDetails';
+import { type SuiObjectResponse } from '@mysten/sui.js/dist/cjs/client';
+
+const PACKAGE_TYPE_NAME = 'Move Package';
 
 function Header({
+	pageType,
 	address,
+	domainName,
 	loading,
 	error,
+	data,
 }: {
+	pageType: 'Package' | 'Object' | 'Address';
 	address: string;
 	loading?: boolean;
+	domainName?: string | null;
 	error?: Error | null;
+	data?: DataType | SuiObjectResponse | null;
 }) {
-	const { data: domainName, isLoading, error: resolveSuinsError } = useResolveSuiNSName(address);
-	const { data, isPending, error: getObjectError } = useGetObject(address!);
-	const isObject = !!data?.data;
-	const errorText = getObjectError?.message ?? resolveSuinsError?.message ?? error?.message;
-
 	return (
 		<div>
 			<PageHeader
-				error={errorText}
-				loading={loading || isLoading || isPending}
-				type={isObject ? 'Object' : 'Address'}
+				error={error?.message}
+				loading={loading}
+				type={pageType}
 				title={address}
 				subtitle={domainName}
 				before={<ObjectDetailsHeader className="h-6 w-6" />}
-				after={<TotalStaked address={address} />}
+				after={
+					pageType === 'Package' &&
+					data &&
+					'id' in data &&
+					'owner' in data &&
+					'version' in data &&
+					'objType' in data ? (
+						<PackageDetails data={data} />
+					) : (
+						<TotalStaked address={address} />
+					)
+				}
 			/>
 
 			<ErrorBoundary>
-				{data && (
+				{data && pageType !== 'Package' && data && 'error' in data && (
```

Comment:
Not sure if `error` will work here. `SuiObjectResponse` based on it's type can be 
```
{
	data?: SuiObjectData | null;
	error?: ObjectResponseError | null;
}
```
So it can only have `data` for example. Maybe checking that `id` is not in `data` would be better